project_files/bank_stripped.js

```diff
@@ -0,0 +1,37 @@
+class Printer {     // Responsibility - Print account history
+    printStatement(account) {
+        console.log("Account Summary:\nDate || Credit || Debit || Balance");
+        account.transactions.forEach(transaction => {
+            console.log(transaction.join(" || "));
+        });
+    }
+}
+
+class Account {     // Responsibility - Store account details (balance and transaction history)
```

Comment:
Here's a good test for whether a class follows the principle single responsibility: you can state the class’s responsibility without any and's, or's, or but's

Refactoring recipe: extract class